scripts/publish-npm.js

```diff
@@ -43,23 +45,28 @@
  * there is a version higher than XY
  *
  * Important tags:
- * If tag v0.XY.0-rcZ is present on the commit then publish to npm with version 0.XY.0-rcZ and tag next
+ * If tag v0.XY.0-rc.Z is present on the commit then publish to npm with version 0.XY.0-rc.Z and tag next
  * If tag v0.XY.Z is present on the commit then publish to npm with version 0.XY.Z and no tag (npm will consider it latest)
  */
 
 /*eslint-disable no-undef */
 require('shelljs/global');
 
-const buildBranch = process.env.CIRCLE_BRANCH;
+const buildTag = process.env.CIRCLE_TAG;
```

Comment:
We are no longer running the `publish_npm_package` job on stable branches, but rather we look for the actual git tag (e.g. v0.57.1, see Circle CI config).

$CIRCLE_TAG is available when a Circle CI job is initiated due to a commit that matches the pattern established in `.circleci/config.yml`.

This envvar is empty when a job is initiated due to a commit to a branch, so we need to switch to using $CIRCLE_TAG instead.